### PR TITLE
Link or download assemblies from IRIDA REST API

### PR DESCRIPTION
**Pull request by tom114**
This PR adds the ability to link or download assemblies from IRIDA's REST API.

This is for https://github.com/phac-nml/irida/issues/121

### CODE REVIEW

File: install/install.pl [hunk 1]
```diff
@@ -42,7 +42,7 @@
 my $iridaDir = $ENV{HOME}."/.irida/";
 my $confFile = "$iridaDir/ngs-archive-linker.conf";
 
-$ret = option("Install config file to $confFile?","y");
+my $ret = option("Install config file to $confFile?","y");
```
Comment (tom114) [inline install/install.pl:45]: This is just to fix a missing variable init in the install script.

Comment (Takadonet) [reply install/install.pl:45]: Sounds good

File: ngsArchiveLinker.pl [hunk 2]
```diff
@@ -275,6 +277,35 @@ sub createLinks{
...
+		my $basename = basename($filename);
+
+		my $newfile = "$sampleDir/$basename";
+
+		if($download){
+			downloadFile($fileref->{href},$newfile,$client,$headers,$duplicateLevel,$type);
+		}
+		else{
+			if(! -e $filename){
+				die "Error: Script cannot see a file to be linked: $filename.  Ensure you have access to the sequence files directory.";
```
Comment (Takadonet) [inline ngsArchiveLinker.pl]: Indicate **assemblies** and not just **sequence files** in error message.

**Review by Takadonet — CHANGES_REQUESTED** [review]

File: ngsArchiveLinker.pl (hunk 2 above)
Comment (tom114) [reply ngsArchiveLinker.pl]: Done in e8b07dc

**Review by Takadonet — APPROVED** [review]
Looks good

**Pull request merged**